test/common/upstream/upstream_impl_test.cc

```diff
@@ -3395,6 +3396,41 @@ TEST_F(ClusterInfoImplTest, Http3BadConfig) {
   EXPECT_THROW_WITH_REGEX(makeCluster(yaml), EnvoyException,
                           "HTTP3 configured but not enabled in the build.");
 }
+TEST_F(ClusterInfoImplTest, UseDownstreamHttpProtocolWithoutDowngrade) {
```

Comment:
I think this may need to go above `#else` at least?

Reply:
Indeed, nice catch :-)